test/notifications/display_test.dart

```diff
@@ -106,26 +108,54 @@ void main() {
   });
 
   group('NotificationDisplayManager show', () {
-    void checkNotification(MessageFcmMessage data, {
+    void checkNotification(List<MessageFcmMessage> messages, {
       required String expectedTitle,
       required String expectedTagComponent,
+      required bool expectedGroup,
     }) {
-      final expectedTag = '${data.realmUri}|${data.userId}|$expectedTagComponent';
-      final expectedGroupKey = '${data.realmUri}|${data.userId}';
-      final expectedId =
-        NotificationDisplayManager.notificationIdAsHashOf(expectedTag);
       const expectedIntentFlags =
         PendingIntentFlag.immutable | PendingIntentFlag.updateCurrent;
 
-      check(testBinding.androidNotificationHost.takeNotifyCalls())
-        ..length.equals(2)
-        ..containsInOrder(<Condition<AndroidNotificationHostApiNotifyCall>>[
-          (it) => it
+      final notifyCallsChecks = <Condition<Object?>>[];
+      for (int i = 0; i < messages.length; i++) {
+        final data = messages[i];
+        final expectedTag =
+          '${data.realmUri}|${data.userId}|$expectedTagComponent';
+        final expectedGroupKey = '${data.realmUri}|${data.userId}';
+        final expectedId =
+          NotificationDisplayManager.notificationIdAsHashOf(expectedTag);
+        final expectedSelfUserKey = '${data.realmUri}|${data.userId}';
+
+        // List of all the checks for messages on each notify calls.
+        final messagesChecks = <Condition<Object?>>[];
+        for (int j = 0; j < (i + 1); j++) {
+          final data = messages[j];
+          final expectedSenderKey = '${data.realmUri}|${data.senderId}';
+          messagesChecks.add((it) => it.isA<MessagingStyleMessage>()
+            ..text.equals(data.content)
+            ..timestampMs.equals(data.time * 1000)
+            ..person.which((it) => it.isNotNull()
+              ..iconBitmap.isNotNull()
```

Comment:
Huh, what does the sender icon look like? I'd think we'd have to set up the test HTTP client for that fetch to not give an error.

(Is it returning an error response or something, and we're getting the bytes of that body? If so then that seems like a bug in the code under test, which we should fix — we only actually want the response body bytes if the HTTP status was 200.)

It'd be good to, for at least one test case, set up the test HTTP client to return a known value at the expected avatar URL, and then check that we get that value.

Reply:
The fake http client seems to return an empty list for `Response.bodyBytes` here, so this turns out to be a non-null empty list.

----

I'll add the fake http client infra in the followup PR with the above mentioned tests.